src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -438,6 +488,8 @@ private static bool CheckConstraintsSingleType(TypeSymbol type, CheckConstraints
             Location location,
             DiagnosticBag diagnostics)
         {
+            type = (NamedTypeSymbol)type.TupleUnderlyingTypeOrSelf();
```

Comment:
>type = (NamedTypeSymbol)type.TupleUnderlyingTypeOrSelf(); [](start = 12, length = 57)

I don't think this is going to be sufficient to handle long tuples properly. Please add a test when violation is after 7th element.